File: backend/app/http/biz/handler/user/user_api.go
package user

import (
	"context"

	"backend/app/http/biz/service"
	"backend/app/http/biz/utils"
	user "backend/app/http/hertz_gen/user"
	"github.com/cloudwego/hertz/pkg/app"
	"github.com/cloudwego/hertz/pkg/protocol/consts"
)

// LoginWithUsername .
// @router /v1/user/login/username [POST]
func LoginWithUsername(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.LoginWithUsernameReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.LoginResp{}
	resp, err = service.NewLoginWithUsernameService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// LoginWithEmail .
// @router /v1/user/login/email [POST]
func LoginWithEmail(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.LoginWithEmailReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.LoginResp{}
	resp, err = service.NewLoginWithEmailService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// LoginWithPhone .
// @router /v1/user/login/phone [POST]
func LoginWithPhone(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.LoginWithPhoneReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.LoginResp{}
	resp, err = service.NewLoginWithPhoneService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// Register .
// @router /v1/user/register [POST]
func Register(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.RegisterReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.LoginResp{}
	resp, err = service.NewRegisterService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// UpdateUserInfo .
// @router /v1/auth/user/info [POST]
func UpdateUserInfo(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.UpdateUserInfoReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.UpdateUserInfoResp{}
	resp, err = service.NewUpdateUserInfoService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// Logout .
// @router /v1/auth/user/logout [POST]
func Logout(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.LogoutReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.LogoutResp{}
	resp, err = service.NewLogoutService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// FollowUser .
// @router /v1/auth/user/follow [POST]
func FollowUser(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.FollowUserReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.FollowUserResp{}
	resp, err = service.NewFollowUserService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// FollowerList .
// @router /v1/user/followers [GET]
func FollowerList(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.FollowerListReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.FollowerListResp{}
	resp, err = service.NewFollowerListService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// FollowingList .
// @router /v1/user/following [GET]
func FollowingList(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.FollowingListReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.FollowingListResp{}
	resp, err = service.NewFollowingListService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// FriendList .
// @router /v1/user/friends [GET]
func FriendList(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.FriendListReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.FriendListResp{}
	resp, err = service.NewFriendListService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// UserUploadFile .
// @router /v1/auth/user/upload [POST]
func UserUploadFile(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.UserUploadFileReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.UserUploadFileResp{}
	resp, err = service.NewUserUploadFileService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

// RefreshToken .
// @router /v1/auth/user/refresh [POST]
func RefreshToken(ctx context.Context, c *app.RequestContext) {
	var err error
	var req user.RefreshTokenReq
	err = c.BindAndValidate(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	resp := &user.RefreshTokenResp{}
	resp, err = service.NewRefreshTokenService(ctx, c).Run(&req)
	if err != nil {
		utils.SendErrResponse(ctx, c, consts.StatusOK, err)
		return
	}

	utils.SendSuccessResponse(ctx, c, consts.StatusOK, resp)
}

File: frontend/src/api/userApi.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utility/request'

/** 关注或取消关注用户 POST /v1/auth/user/follow */
export async function followUser(body: API.FollowUserReq, options?: { [key: string]: any }) {
    return request<API.FollowUserResp>('/v1/auth/user/follow', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 更新用户信息 POST /v1/auth/user/info */
export async function updateUserInfo(body: API.UpdateUserInfoReq, options?: { [key: string]: any }) {
    return request<API.UpdateUserInfoResp>('/v1/auth/user/info', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 用户登出 POST /v1/auth/user/logout */
export async function logout(body: API.LogoutReq, options?: { [key: string]: any }) {
    return request<API.LogoutResp>('/v1/auth/user/logout', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 刷新令牌 POST /v1/auth/user/refresh */
export async function refreshToken(options?: { [key: string]: any }) {
    return request<API.RefreshTokenResp>('/v1/auth/user/refresh', {
        method: 'POST',
        ...(options || {})
    })
}

/** 获取粉丝列表 GET /v1/user/followers */
export async function followerList(
    // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
    params: API.FollowerListParams,
    options?: { [key: string]: any }
) {
    return request<API.FollowerListResp>('/v1/user/followers', {
        method: 'GET',
        params: {
            ...params
        },
        ...(options || {})
    })
}

/** 获取关注列表 GET /v1/user/following */
export async function followingList(
    // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
    params: API.FollowingListParams,
    options?: { [key: string]: any }
) {
    return request<API.FollowingListResp>('/v1/user/following', {
        method: 'GET',
        params: {
            ...params
        },
        ...(options || {})
    })
}

/** 获取好友列表 GET /v1/user/friends */
export async function friendList(
    // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
    params: API.FriendListParams,
    options?: { [key: string]: any }
) {
    return request<API.FriendListResp>('/v1/user/friends', {
        method: 'GET',
        params: {
            ...params
        },
        ...(options || {})
    })
}

/** 邮箱登录 POST /v1/user/login/email */
export async function loginWithEmail(body: API.LoginWithEmailReq, options?: { [key: string]: any }) {
    return request<API.LoginResp>('/v1/user/login/email', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 手机号登录 POST /v1/user/login/phone */
export async function loginWithPhone(body: API.LoginWithPhoneReq, options?: { [key: string]: any }) {
    return request<API.LoginResp>('/v1/user/login/phone', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 用户名登录 POST /v1/user/login/username */
export async function loginWithUsername(body: API.LoginWithUsernameReq, options?: { [key: string]: any }) {
    return request<API.LoginResp>('/v1/user/login/username', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}

/** 用户注册 POST /v1/user/register */
export async function register(body: API.RegisterReq, options?: { [key: string]: any }) {
    return request<API.LoginResp>('/v1/user/register', {
        method: 'POST',
        headers: {
            'Content-Type': 'application/json'
        },
        data: body,
        ...(options || {})
    })
}
